homeassistant/components/qnap_qsw/sensor.py

```diff
@@ -0,0 +1,134 @@
+from .coordinator import QswUpdateCoordinator
+
+
+@dataclass
+class QswSensorEntityDescription(SensorEntityDescription):
+    """A class that describes QNAP QSW sensor entities."""
+
+    attributes: dict[str, list[str]] | None = None
+    subkey: str | None = None
+
+
+SENSOR_TYPES: Final[tuple[QswSensorEntityDescription, ...]] = (
+    QswSensorEntityDescription(
+        key=QSD_SYSTEM_SENSOR,
+        name="Fan 1 Speed",
+        native_unit_of_measurement=RPM,
+        state_class=SensorStateClass.MEASUREMENT,
+        subkey=QSD_FAN1_SPEED,
+    ),
+    QswSensorEntityDescription(
+        key=QSD_SYSTEM_SENSOR,
+        name="Fan 2 Speed",
+        native_unit_of_measurement=RPM,
+        state_class=SensorStateClass.MEASUREMENT,
+        subkey=QSD_FAN2_SPEED,
+    ),
+    QswSensorEntityDescription(
+        attributes={
+            ATTR_MAX: [QSD_SYSTEM_SENSOR, QSD_TEMP_MAX],
+        },
+        device_class=SensorDeviceClass.TEMPERATURE,
+        key=QSD_SYSTEM_SENSOR,
+        name="Temperature",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        subkey=QSD_TEMP,
+    ),
+    QswSensorEntityDescription(
+        key=QSD_SYSTEM_TIME,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        name="Uptime",
+        native_unit_of_measurement=TIME_SECONDS,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        subkey=QSD_UPTIME,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add QNAP QSW sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for description in SENSOR_TYPES:
+        if (
+            description.key in coordinator.data
+            and description.subkey in coordinator.data[description.key]
+        ):
+            sensors.append(QswSensor(coordinator, description, entry))
+
+    async_add_entities(sensors)
+
+
+class QswSensor(QswEntity, SensorEntity):
+    """Define a QNAP QSW sensor."""
+
+    def __init__(
+        self,
+        coordinator: QswUpdateCoordinator,
+        description: QswSensorEntityDescription,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, entry)
+        self._attr_name = (
+            f"{self.get_entity_value(QSD_SYSTEM_BOARD, QSD_PRODUCT)} {description.name}"
+        )
+        self._attr_unique_id = (
+            f"{entry.entry_id}_{description.key}_{description.subkey}"
+        )
+        self.attributes = description.attributes
+        self.entity_description = description
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return state attributes."""
+        if not self.attributes:
+            return None
+        return {
+            key: self.get_entity_value(val[0], val[1])
+            for key, val in self.attributes.items()
+        }
```

Comment:
What are these attributes?
https://developers.home-assistant.io/docs/core/entity/?_highlight=extra_state_attributes
Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additional sensor entities.

Reply:
In this case the only extra attribute is the maximum temperature, which is a constant value (85ºC for my switch) and never changes.